src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignerInfo.cs

```diff
@@ -663,23 +667,40 @@ private static bool VerifyAttributes(
 
             if (compatMode)
             {
-                byte[] encoded = writer.Encode();
-                encoded[0] = 0x31;
-                return verify(encoded);
+                int encodedLength = writer.GetEncodedLength();
+                byte[]? rented = null;
+                Span<byte> encoded = encodedLength <= 256
```

Comment:
the right pattern is to always check both lower and upper bound, so please add (uint) cast here

Reply:
The input into `stackalloc` is a const. Worse case the negative value goes in to the pool rental which will throw.

What are we protecting against with the cast?

Reply:
Ah, yes, ignore my comment, I didn't notice the stackalloc is constant-sized